### ISSUE TITLE
"unexpected token" when using | pattern in matches! macro

### ISSUE THREAD

**Issue by iitalics** [rust-lang/rust-analyzer#15056]
i get "unexpected token in input" error on the `matches!` invocation with the following file contents:

```rust
pub fn is_paren(ch: char) -> bool {
    matches!(ch, '(' | ')')
}
```

this is valid syntax: cargo check/build etc. work fine

**rust-analyzer version**: (eg. output of "rust-analyzer: Show RA Version" command, accessible in VSCode via <kbd>Ctrl/⌘</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>)

`rust-analyzer 1.70.0 (90c5418 2023-05-31)`

**rustc version**: (eg. output of `rustc -V`)

`rustc 1.70.0 (90c541806 2023-05-31)`

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

using rustic emacs

**Comment by iitalics** [comment]
if i add additional parenthesis then rust-analyzer accepts it:
```rust
pub fn is_paren(ch: char) -> bool {
    matches!(ch, ('(' | ')'))
}
```

**Comment by Veykril** [comment]
The fix didn't make it to the rustup component right ... 

This is already fixed on master, the 1.70 rustup component doesn't have the fix though so you'll have to change the server you are using.
